akka-docs-dev/rst/scala/code/docs/stream/FlowDocSpec.scala

```diff
@@ -138,4 +138,69 @@ class FlowDocSpec extends AkkaSpec {
 
     //#flow-connecting
   }
+
+  "various ways of transforming materialized values" in {
+    import scala.concurrent.duration._
+
+    val throttler = Flow(Source(1.second, 1.second, "test")) { implicit b =>
+      tickSource =>
+        import FlowGraph.Implicits._
+        val zip = b.add(Zip[String, Int](OperationAttributes.inputBuffer(1, 1)))
+
+        (zip.in1, zip.out.map(_._2).outlet)
+    }
+
+    //#flow-mat-combine
+    // An empty source that can be shut down explicitly from the outside
+    val source: Source[Int, Promise[Unit]] = Source.lazyEmpty[Int]()
+
+    // A flow that internally throttles elements to 1/second, and returns a Cancellable
+    // which can be used to shut down the stream
+    val flow: Flow[Int, Int, Cancellable] = throttler
+
+    // A sink that returns the first element of a stream in the returned Future
+    val sink: Sink[Int, Future[Int]] = Sink.head[Int]()
+
+    // Simple selection of materialized values by using Keep.right
```

Comment:
perhaps say that the default of `via` and `to` is to keep the left value—even if it is in the text, this might bear repeating